### PR TITLE
[Queue] Work queue Skeleton MVP

### PR DESCRIPTION
**Pull request by dannysellers**
Implements [#1752](https://github.com/department-of-veterans-affairs/appeals-pm/pull/1752), #4254.

### CODE REVIEW

File: client/app/queue/QueueListView.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        </div>
+      </div>
+    </div>;
+  }
+}
+
+QueueListView.propTypes = {};
+
+const mapStateToProps = (state, props) => {
+  return {
```
Comment (nickheiner-usds) [inline client/app/queue/QueueListView.jsx]: Please use the `() => expression` syntax instead of `return` when you're only returning a single value.

File: client/app/queue/QueueListView.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import QueueTable from './QueueTable';
+import StatusMessage from '../components/StatusMessage';
+
+class QueueListView extends React.Component {
```
Comment (nickheiner-usds) [inline client/app/queue/QueueListView.jsx]: Please default to extending `PureComponent`.

File: client/app/queue/QueueLoadingScreen.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        const { appeals, tasks } = returnedObject;
+
+        this.props.onReceiveQueue(appeals, tasks);
+      });
+  }
+
+  render = () => {
+    const failStatusMessageChildren = <div>
+      It looks like Caseflow was unable to load this case.<br />
+      Please <a href="javascript:void(0)" onClick={() => location.reload()}>refresh the page</a> and try again.
```
Comment (nickheiner-usds) [inline client/app/queue/QueueLoadingScreen.jsx]: Why is this necessary?

```
href="javascript:void(0)"
```

File: client/app/queue/index.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReduxBase from '@department-of-veterans-affairs/appeals-frontend-toolkit/components/ReduxBase';
```
Comment (nickheiner-usds) [inline client/app/queue/index.jsx]: 👍 Thanks!

**Comment by nickheiner-usds** [comment]
I left a few comments before I saw it was still WIP. :smile: Hopefully I wasn't jumping the gun too much. Can you add me as a reviewer when it's ready?

File: app/assets/stylesheets/_main.scss [hunk 5]
```diff
@@ -234,6 +235,48 @@ body {
   background-size: 32px 32px;
 }
 
+.cf-logo-image-reader {
```
Comment (dannysellers) [inline app/assets/stylesheets/_main.scss]: These were all deleted in `master`, but restored during the merge.. 👍 fixing

File: client/app/components/Highlight.jsx [hunk 6]
```diff
@@ -27,9 +27,17 @@ Highlight.defaultProps = {
   children: ''
 };
 
-const mapStateToProps = (state) => ({
-  searchQuery: state.documentList.docFilterCriteria.searchQuery
-});
+const mapStateToProps = (state, ownProps) => {
+  const props = _.extend({}, ownProps);
```
Comment (nickheiner-usds) [inline client/app/components/Highlight.jsx]: This can just be: `_.clone`

**Comment by nickheiner-usds** [comment]
Note from in-person conversation: reminder to enable `prefer-const` in Queue directory.

File: client/app/queue/QueueLoadingScreen.jsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    const failStatusMessageChildren = <div>
+      It looks like Caseflow was unable to load this case.<br />
+      Please <a onClick={this.reload}>refresh the page</a> and try again.
+    </div>;
+
+    const loadingDataDisplay = <LoadingDataDisplay
+      createLoadPromise={this.createLoadPromise}
+      loadingScreenProps={{
+        spinnerColor: Constants.QUEUE_COLOR,
+        message: 'Loading claims folder in Reader...'
```
Comment (nickheiner-usds) [inline client/app/queue/QueueLoadingScreen.jsx]: Is `'Loading claims folder in Reader...'` really the right message?

File: client/app/queue/index.jsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+import QueueManager from './QueueManager';
+
+import rootReducer from './reducers';
+
+export default class WorkQueue extends React.Component {
```
Comment (nickheiner-usds) [inline client/app/queue/index.jsx]: Let's either make this a `PureComponent`, or if it's just a `render` function, make it a functional component.

File: client/app/queue/reducers.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
+import { timeFunction } from '../util/PerfDebug';
+import { update } from '../util/ReducerUtil';
+import * as Constants from './actionTypes';
+
+export const initialState = {
+  loadedQueueId: null,
+  didLoadQueueFail: false,
```
Comment (nickheiner-usds) [inline client/app/queue/reducers.js]: I don't see where we're reading this value. We can add it back in in later PRs, but I'd like to have only code that's actually being used enter `master`.

Comment (nickheiner-usds) [reply client/app/queue/reducers.js]: (Same comment applies to some other fields in this object.)

File: client/app/queue/actionTypes.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export const SET_LOADED_QUEUE_ID = 'SET_LOADED_QUEUE_ID';
+export const RECEIVE_QUEUE_DETAILS = 'RECEIVE_QUEUE_DETAILS';
+export const RECEIVE_QUEUE_DETAILS_FAILURE = 'RECEIVE_QUEUE_DETAILS_FAILURE';
+
+export const SHOW_SEARCH_BAR = 'SHOW_SEARCH_BAR';
```
Comment (nickheiner-usds) [inline client/app/queue/actionTypes.js]: Looks like this is not being used yet. Let's remove it. We can add it back later.

File: client/app/queue/QueueManager.jsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+          defaultUrl="/"
+          userDisplayName={this.props.userDisplayName}
+          dropdownUrls={this.props.dropdownUrls}
+          logoProps={{
+            backgroundColor: Constants.QUEUE_LOGO_BACKGROUND_COLOR,
+            overlapColor: Constants.QUEUE_LOGO_OVERLAP_COLOR,
+            accentColor: Constants.QUEUE_COLOR
+          }}
+          appName="Queue">
+          <div className="cf-wide-app section--queue-list">
```
Comment (nickheiner-usds) [inline client/app/queue/QueueManager.jsx]: I do not see this class being used.

**Comment by nickheiner-usds** [comment]
Beautiful! :smile:

image

File: client/app/queue/reducers.js (hunk 9 above)
Comment (dannysellers) [reply client/app/queue/reducers.js]: Speaking of code that's actually being used, we should probably add a FeatureToggle for access to `/queue` @amprokop

File: client/app/components/Highlight.jsx [hunk 12]
```diff
@@ -27,9 +27,17 @@ Highlight.defaultProps = {
   children: ''
 };
 
-const mapStateToProps = (state) => ({
```
Comment (nickheiner-usds) [inline client/app/components/Highlight.jsx]: I have a few questions about this approach, but this change isn't needed for this PR, so let's just `git checkout master -- client/app/components/Highlight.jsx` and deal with it when it's needed.

File: app/views/queue/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<% content_for :full_page_content do %>
+  <%= react_component("BaseContainer", props: {
```
Comment (nickheiner-usds) [inline app/views/queue/index.html.erb]: We only need `BaseContainer` if we are going to be using its alert functionality, but I don't see us doing that. (We are not using the `handleAlert` prop it passes in.) Instead, you should follow the pattern that Hearings Prep does, where it skips `BaseContainer` and renders onto the page directly.

We have a few other apps that use `BaseContainer` for no reason. This is just a mistake.

File: client/app/queue/QueueLoadingScreen.jsx [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+import ApiUtil from '../util/ApiUtil';
+import LoadingDataDisplay from '../components/LoadingDataDisplay';
+import * as Constants from './constants';
+
+class QueueLoadingScreen extends React.PureComponent {
+  getUserId = () => 1;
+
+  createLoadPromise = () => {
+    // todo: Promise.resolve() if appeals/tasks already loaded
+    // todo: return ApiUtil.get(`/queue/${this.props.userId}`, {}).
```
Comment (nickheiner-usds) [inline client/app/queue/QueueLoadingScreen.jsx]: This comment doesn't seem like it needs to be here.

File: client/app/queue/QueueListView.jsx [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+      </StatusMessage>;
+    } else {
+      tableContent = <div>
+        <h1 className="cf-push-left">Your Queue</h1>
+        <span>{JSON.stringify(this.props.tasks)}</span><br />
+        <span>{JSON.stringify(this.props.appeals)}</span>
+      </div>;
+    }
+
+    return <div className="usa-grid">
```
Comment (nickheiner-usds) [inline client/app/queue/QueueListView.jsx]: The way we have this set up now, we nest `usa-grid` and `cf-app` twice. I'm pretty sure this is unnecessary.

image

Also, I'm pretty sure that `cf-app` isn't actually doing anything.

File: app/models/serializers/work_queue/appeal_serializer.rb [hunk 16]
```diff
@@ -28,6 +28,8 @@ class WorkQueue::AppealSerializer < ActiveModel::Serializer
   attribute :vacols_id
   attribute :type
   attribute :aod
+  attribute :docket_number
```
Comment (nickheiner-usds) [inline app/models/serializers/work_queue/appeal_serializer.rb]: What is this for? Can we remove it if it's unused?

Comment (dannysellers) [reply app/models/serializers/work_queue/appeal_serializer.rb]: It is/will be used in the `QueueTable`, but sure

File: client/app/queue/QueueLoadingScreen.jsx (hunk 14 above)
Comment (dannysellers) [reply client/app/queue/QueueLoadingScreen.jsx]: It's a reminder that the GET run in this part (which currently relies on `this.getUserId`, which just returns 1) is a stub, to be replaced by a real `userId` passed from elsewhere/a higher component

File: app/views/queue/index.html.erb (hunk 13 above)
Comment (dannysellers) [reply app/views/queue/index.html.erb]: So we'll never need this alert functionality for the new queue stuff?

Comment (nickheiner-usds) [reply app/views/queue/index.html.erb]: Let's assume that we won't. It's easy enough to refactor if we ever do.
